deployment/cdk/opensearch-service-migration/lib/stack-composer.ts

```diff
@@ -222,11 +275,11 @@ export class StackComposer {
         if (addOnMigrationDeployId && vpcId) {
             console.warn("Addon deployments will use the original deployment 'vpcId' regardless of passed 'vpcId' values")
         }
-        let targetEndpoint
-        if (targetClusterEndpoint && osContainerServiceEnabled) {
-            throw new Error("The following options are mutually exclusive as only one target cluster can be specified for a given deployment: [targetClusterEndpoint, osContainerServiceEnabled]")
-        } else if (targetClusterEndpoint || osContainerServiceEnabled) {
-            targetEndpoint = targetClusterEndpoint ? targetClusterEndpoint : "https://opensearch:9200"
+        let preexistingOrContainerTargetEndpoint
+        if (targetCluster && osContainerServiceEnabled) {
```

Comment:
nit: reads strange that this block specifies targetCluster, but then uses targetClusterEndpoint instead of targetCluster?.endpoint